player/lua/console.lua

```diff
@@ -911,18 +911,14 @@ function build_completers()
 end
 
 -- Find the longest common case-sensitive prefix of the entries in "list".
-local function find_common_prefix(part, list)
-    local prefix = nil
-
-    for _, candidate in ipairs(list) do
-        if prefix and prefix ~= candidate then
-            local prefix_len = part:len()
-            while prefix:sub(1, prefix_len + 1) == candidate:sub(1, prefix_len + 1) do
-                prefix_len = prefix_len + 1
-            end
-            prefix = candidate:sub(1, prefix_len)
-        else
-            prefix = candidate
+local function find_common_prefix(list)
+    local prefix = list[1]
+
+    for i = 2, #list do
+        local common_length = common_prefix_length(prefix, list[i])
+
+        if common_length < #prefix then
+            prefix = prefix:sub(1, common_length)
```

Comment:
Have you ever tried doing this without the condition? I'd imagine that lua simply returns the same string without much overhead.

Reply:
Yeah it's about the same speed.

Reply:
So set \<Tab> is the same speed while set s\<Tab> goes from 6e-7s to 5e-6s but since this is super fast anyway I'll go with the shorter version.

Reply:
Nevermind, the measured time changes often and I had to increase the number of iterations. With 10k iterations these are the average times I get:

old version:
set \<Tab>: 1e-4s
set s\Tab>: 1e-5s

new version with intermediate variable:
set \<Tab>: 7e-5s
set s\<Tab>: 6e-7s

new version always calling :sub():
set \<Tab>: 1e-5s
set s\<Tab>: 5e-6s

So this is both the shortest and faster version.